### PR TITLE
[plugin.audio.cbcradio] v2.0.4

### PR DESCRIPTION
**Pull request by ihemingway**
### Description
Plugin for steaming live stations from CBC Radio One and CBC Music
### Checklist:
- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.
- If you see no activity on your PR after a week (so at least one weekend has passed) then please go to the #kodi-dev freenode IRC channel to reach out to the team

### CODE REVIEW

File: plugin.audio.cbcradio/addon.xml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+<addon id="plugin.audio.cbcradio" name="CBC Radio" version="2.0.4" provider-name="ihemingway">
+    <requires>
+        <import addon="xbmc.python" version="3.0.0" />
+    </requires>
+    <extension point="xbmc.python.pluginsource" library="default.py">
+        <provides>audio</provides>
+    </extension>
+    <extension point="xbmc.addon.metadata">
+        <summary lang="en_GB">CBC Radio</summary>
+        <license>GNU General Public License, v2</license>
```
Comment (basrieter) [inline plugin.audio.cbcradio/addon.xml]: Please use the SPDX identifier (https://spdx.org/licenses/) and make sure the license matches the license from the LICENSE.md. It currentlty does not!

File: plugin.audio.cbcradio/default.py [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+def play_stream(key, location, url):
+    program_schedule, program, playlog, last_track, play_item = initialize(key, location, url)
+    c = -1
+    while not MONITOR.abortRequested():
+        now = time.time() * 1000
+        if calc_minutes() in [00, 1, 2, 3, 4, 5] and key == 2:
+        # cbc music has 5 min news breaks at the top of the hour
+            news_break()
+            while calc_minutes() in [00, 1, 2, 3, 4, 5]:
+                xbmc.sleep(10000)
```
Comment (basrieter) [inline plugin.audio.cbcradio/default.py]: This call completely block execution for 10 seconds without the option to abort. You should also use the `MONITOR.waitForAbort()` here and test for abort afterwards. Just like in line 198

File: plugin.audio.cbcradio/addon.xml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
```
Comment (basrieter) [inline plugin.audio.cbcradio/addon.xml:1]: Please include language info via the `<language>` tag with the 2-letter language code.

**Review by basrieter — CHANGES_REQUESTED** [review]
Please see my comments.

**Comment by basrieter** [comment]
Also make the add-on checker happy: 
image

File: plugin.audio.cbcradio/addon.xml (hunk 1 above)
Comment (ihemingway) [reply plugin.audio.cbcradio/addon.xml]: done

File: plugin.audio.cbcradio/default.py (hunk 2 above)
Comment (ihemingway) [reply plugin.audio.cbcradio/default.py]: done

File: plugin.audio.cbcradio/addon.xml (hunk 3 above)
Comment (ihemingway) [reply plugin.audio.cbcradio/addon.xml:1]: done

File: plugin.audio.cbcradio/cbc.py [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+                pass
+            else:
+                tag.setAlbum(None)
+                tag.setArtist(program.host)
+                tag.setTitle(program.title)
+                tag.setComment(None)
+                PLAYER.updateInfoTag(play_item)
+        chill(5)
+        if MONITOR.abortRequested():
+            abort()
```
Comment (ihemingway) [inline plugin.audio.cbcradio/cbc.py]: ah crap.... was going to do an abort func and then changed it to chill()
i'll fix this tomorrow. i literally just have to delete lines 212-13.

Comment (ihemingway) [reply plugin.audio.cbcradio/cbc.py]: i guess have a look anyway and let me know if there is anything besides this to fix before i resubmit.

Comment (ihemingway) [reply plugin.audio.cbcradio/cbc.py]: actually, it should technically never reach the non-existent func. the check for `MONITOR.abortRequested()` is in the `chill()` func.... testing on my dev box i never get an exception when exiting.

up to you if you want it removed or not.

**Comment by ihemingway** [comment]
removed those two lines. should be fine now unless there are further problems

**Review by basrieter — APPROVED** [review]

**Pull request merged**